File: leetcode/0008/myAtoi.py
def myAtoi(s: str) -> int:
    # 移除前导空格
    s = s.strip()
    flag = 1
    res = ''
    for i in range(len(s)):
        # 检查首字符是否为符号
        if i == 0 and s[i] in ['-', '+']:
            flag = -1 if s[i] == '-' else 1
            continue
        # 判断是否是数字
        if s[i].isdigit():
            res += s[i]
        else:
            break
    if res == '':
        res = 0
    res = flag * int(res)
    return min(res, 2**31 - 1) if flag == 1 else max(res, -2**31)

print(myAtoi('-123'))

File: leetcode/0008/myAtoi.swift
func myAtoi(_ s: String) -> Int {
    var sign = 1
    var res = 0
    var foundDigit = false
    var foundSign = false

    for char in s {
        if char == " " {
            if foundDigit || foundSign {
                break
            }
            continue
        }
        if char == "+" || char == "-" {
            if foundDigit || foundSign {
                break
            }
            foundSign = true
            sign = char == "-" ? -1 : 1
            continue
        }
        guard let n = char.wholeNumberValue else {
            break
        }
        foundDigit = true
        if res * 10 + n > Int(Int32.max) {
            res = Int(Int32.max)
            res += sign == 1 ? 0 : 1
            break
        }
        res = res * 10 + n
    }
    return res * sign
}
